statistics/histogram.go

```diff
@@ -395,10 +395,46 @@ func validRange(sc *stmtctx.StatementContext, ran *ranger.Range, encoded bool) b
 	return bytes.Compare(low, high) < 0
 }
 
+func checkKind(vals []types.Datum, kind byte) bool {
+	if kind == types.KindString {
+		kind = types.KindBytes
+	}
+	for _, val := range vals {
```

Comment:
need we loop all values? what abouting only checking the first not null value?

Reply:
Done.